File: DistributTools/DataModel/AppItemModel.m
//
//  AppItemModel.m
//  PuhuiDownloader
//
//  Created by zhangyinglong on 16/1/9.
//  Copyright © 2016年 普惠金融. All rights reserved.
//

#import "AppItemModel.h"

@implementation AppItemModel

- (BOOL)isNewVersion {
    id old = [[NSUserDefaults standardUserDefaults] objectForKey:self.appUpdateModel.appIdentifier];
    if ( old == nil ) {
        return YES;
    } else if ( self.appUpdateModel.appBuildVersion > [old integerValue] ) {
        return YES;
    }
    return NO;
}

@end

File: DistributTools/DataModel/AppItemModel.swift
//
//  AppItemModel.swift
//  DistributTools
//
//  Created by zhang yinglong on 2017/3/7.
//  Copyright © 2017年 ChinaHR. All rights reserved.
//

import ObjectMapper
import AlamofireObjectMapper
import RealmSwift
import SwiftyUserDefaults

class AppItemModel: Object, Mappable, IdentifyProtocol {
    
    var id: String = ""
    var appUpdateModel: AppUpdateModel? = nil;
    var image: UIImage? = nil
    
    override var description: String {
        return ""
//        return "AppItemModel[id=\(id),name=\(name),code=\(code),url=\(url)]"
    }
    
    required convenience init?(map: Map) {
        self.init()
    }
    
    func mapping(map: Map) {
        id <- map["id"]
        appUpdateModel <- map["AppUpdateModel"]
        image <- map["image"]
    }
    
    func setNeedUpdate(need: Bool) {
        
    }
    
}
